holoviews/ipython/magics.py

```diff
@@ -364,74 +375,73 @@ def update_options(cls, options, items):
         Switch default options and backend if new backend
         is supplied in items.
         """
-        backend = items.get('backend', '')
+        backend = items.get('backend', Store.current_backend)
         prev_backend = Store.current_backend
-        renderer = Store.renderers[Store.current_backend]
+        renderer = Store.renderers[prev_backend]
         prev_backend += ':%s' % renderer.mode
 
+        if 'backend' not in options:
+            options['backend'] = backend
+
+        for p in ['fig', 'holomap']:
+            cls.allowed[p] = list_formats(p, backend)
+
         available = backend in Store.renderers.keys()
         if (not backend) or (not available) or backend == prev_backend:
             return options
 
-        cls._backend_options[prev_backend] = cls.options
-
-        backend_options = cls._backend_options[backend]
-        for p in ['fig', 'holomap']:
-            opts = list_formats(p, backend)
-            cls.allowed[p] = opts
-            cls.defaults[p] = opts[0]
-            if p not in backend_options:
-                backend_options[p] = opts[0]
+        backend_options = dict(cls._backend_options[backend])
+        cls._backend_options[prev_backend] = {k: v for k, v in cls.options.items()
+                                              if k in cls.remembered}
 
         backend = backend.split(':')[0]
-        render_params = ['fig', 'holomap', 'size', 'fps', 'dpi', 'css']
-        for p in render_params:
-            if p in backend_options:
-                opt = backend_options[p]
-                cls.defaults[p] = opt
-            else:
-                opt = cls.defaults[p]
-                backend_options[p] = opt
-
         for opt in options:
             if opt not in backend_options:
                 backend_options[opt] = options[opt]
 
-        cls.set_backend(backend)
+        for opt in cls.remembered:
+            if opt not in backend_options:
+                backend_options[opt] = cls.defaults[opt]
+
         return backend_options
 
 
     @classmethod
     def initialize(cls, backend_list):
         cls.backend_list = backend_list
-        backend = cls.options.get('backend', cls.defaults['backend'])
+        backend = cls.options.get('backend', Store.current_backend)
         if backend in Store.renderers:
-            cls.options = dict(cls.defaults)
-            cls._set_render_options(cls.defaults)
-            cls.set_backend(backend)
+            cls.options = dict({k: cls.defaults[k] for k in cls.remembered})
+            cls._set_render_options({}, backend)
```

Comment:
Why do you need to call ``_set_render_options`` if there are no options? I suppose there is state being initialize there?